### DISCUSSION TITLE
model naming conventions dictionary

### DISCUSSION THREAD

**Discussion by stephanclaus** [dbt-labs/dbt-core#5290] [Ideas]
Hi everyone,

I see with great interest that the community thinks more about namespaces and model naming conventions lately: (e.g. here https://docs.getdbt.com/blog/on-the-importance-of-naming). 

One idea to allow enforcing naming conventions would be to implement for a "model name component dictionary" (naming things is hard) where

- the log level (information/warning/error)
- a style can be selected (e.g. snake case)
- the components of the model name string are checked against a _dictionary_ and warns/error if the selected name is not compliant (where the dictionary can be shared between different dbt projects)
- more complex rules are defined (e.g. you cannot have a one character long substring in your model name, except v_ for views etc.

Checking these rules would simply be added to the dbt compile/run command. 

Of course you could build something like this somehow with pre hooks already, but having this even more baked in the standard functionality sounds more appealing to me. 

I am happy for any ideas and discussions here. I am happy to add this as an issue if more people are interested in it.

Cheers
Stephan

**Comment by jeremyyeo** [comment]
This sounds great - and perhaps we should go one step further with prescribing some style defaults (e.g. `snake_case_alpha_num`) in code (rather than in documentation) (the number of times one has come across projects with model/table names that are <ins>mix cased</ins>, <ins>occasionally quoted</ins>, and <ins>contains special characters</ins> (pick any 4 from that list) is too great).

Have a previous example of doing this via hooks as mentioned: https://gist.github.com/jeremyyeo/5e3bae4e7a2ff6c6b554880c0d8d0e86

**Reply by stephanclaus** [reply]
Hi Jeremy, thank you for the comment and the hook solution. 

Just a quick confirmation on your style default - I mentioned it as a second bullet point, do you envision anything on top besides checking for snake_case syntax when thinking about examples you witnessed?

Thank you
Stephan
